Lesson4.ipynb

```diff
@@ -29,6 +32,655 @@
+    "    for row_index,row in enumerate(readCSV):                                           # \"for\" loop statement to fetch the data                     \n",
+    "        if row_index > 0:                                                              # Rows start from 1\n",
+    "            totalSteps = 0                                                             # Define a parameter to store each attendees' total steps\n",
+    "            for column_index, steps_in_column in enumerate(row):                       # Calculate the total steps by using \"for\" loop statement\n",
+    "                if column_index > 3 and steps_in_column != '':                         # Starts from cloumn 4\n",
+    "                    totalSteps = totalSteps + int(steps_in_column)                 \n",
+    "            print(\"Total steps of Member\" + str(row_index) + \" is:\" + str(totalSteps)) # After \"for\" statement completes running, totalSteps is the answer we want"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From the code block above, we get total steps for every attendee, and print them one by one.\n",
+    "\n",
+    "I think somebody have noticed something, when calculating the total steps, we use a new thing in the code on Line 10\n",
+    "\n",
+    "```python \n",
+    "int(step)\n",
+    "```\n",
+    "** **\n",
+    "\n",
+    "Last week, we know str(), which used for converting other types to string. Now I'm going to talk about int().\n",
+    "\n",
+    "When fetching the data from a table, the data types are the same -- All are string type.\n",
+    "\n",
+    "Let's check the code block again.\n",
+    "\n",
+    "```python \n",
+    "totalSteps = 0\n",
+    "    for column_index, steps_in_column in enumerate(row):  \n",
+    "      if column_index > 3 and steps_in_column != '':                                \n",
+    "               totalSteps = totalSteps + int(steps_in_column)\n",
+    "```\n",
+    "** **\n",
+    "\n",
+    "We define totalSteps before adding and set initialization to 0. It is an integer.\n",
+    "\n",
+    "Meanwhile, steps_in_column is the data from table, which is string.\n",
+    "\n",
+    "Since string type can't be used for computing, we need to convert the string to integer if you want to do addition here. \n",
+    "\n",
+    "As a result, **int() function is used for converting other types to integer.** We have to do it in computing."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Redo Lesson 3 Questions\n",
+    "\n",
+    "#### Print names of people with steps count more than 100k/300k/400k\n",
+    "\n",
+    "In lesson 3, we have already known how to print the names of people who clock the steps more than 100k, but the data is in the List which prepared before computing. Now let's use the data from a CSV file to answer the same questions from last lesson.\n",
```

Comment:
Remove reference to lesson 3 since this was an optional challenge question (as discussed with Trung)